### PR TITLE
Media players

### PR DESCRIPTION
**Pull request by juandans01**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
   "dependencies": {
     "@fontsource/open-sans": "^4.5.1",
     "babel-plugin-react-require": "^3.1.3",
+    "moment": "^2.29.1",
```
Comment (gmarcos87) [inline package.json:53]: In the project there is already date-fns as a dependency for time manage, it is a tool that solves many of the problems that momentjs has. Even momentjs recommends its use. https://momentjs.com/docs/#/-project-status/

File: packages/ui/components/AudioView/AudioView.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    fileRef.current.addEventListener('canplay', () => {
+      handleCanPlay(true)
+    })
+
+    fileRef.current.addEventListener("timeupdate", () => {
+      handlePercentage((fileRef.current.currentTime / fileRef.current.duration) * 100)
+    })
+  }, [])
```
Comment (gmarcos87) [inline packages/ui/components/AudioView/AudioView.tsx:27]: If an event is added it should be removed in the function that returns useEffect. Something like:
```suggestion
  return () => fileRef.current.removeEventListener("timeupdate", handleTimeUpdate);
  }, [])
```

File: packages/ui/components/ImageView/ImageView.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+type Props = {
+  file: IReportFile,
+  onClick?: () => void
+};
+
+export const ImageView: FunctionComponent<Props> = ({ file, onClick }) => {
+  return (
+    <div 
+      onClick={onClick}
+      className="w-full h-full bg-gray-25 rounded-md cursor-pointer"
```
Comment (gmarcos87) [inline packages/ui/components/ImageView/ImageView.tsx:13]: You can remove some of the styling options and use tailwind's generic ones.
```suggestion
      className="w-full h-full bg-gray-25 rounded-md cursor-pointer bg-cover bg-center"
```

File: packages/ui/components/MediaControls/MediaButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
+import { FunctionComponent, useState, useEffect, useRef } from 'react'
+import cn from 'classnames'
+import { BsPlus } from "@react-icons/all-files/bs/BsPlus"
+import { BsDash } from "@react-icons/all-files/bs/BsDash"
+import { BsArrowsFullscreen } from "@react-icons/all-files/bs/BsArrowsFullscreen"
+
```
Comment (gmarcos87) [inline packages/ui/components/MediaControls/MediaButtons.tsx:6]:
```suggestion
```

File: packages/ui/components/MediaControls/MediaButtons.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  volumePercentage,
+  muted,
+  toggleMuted,
+  toggleVolume
+}) => {
+  
+  const volumeRef = useRef<HTMLDivElement | null>(null)
+  const [volumeIcon, handleVolumeIcon] = useState(null)
+
+  const getVolumeIcon = () => {
```
Comment (gmarcos87) [inline packages/ui/components/MediaControls/MediaButtons.tsx:43]: This should be a separate component that receives muted and volumePercentage as props. That way you also avoid storing the component as if it were state.

File: packages/ui/components/MediaControls/MediaButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+  useEffect(() => {
+    getVolumeIcon()
+  }, [muted, volumePercentage])
+
+
+
+  return (
+    <div className="h-10 relative flex justify-center items-center">
+
```
Comment (gmarcos87) [inline packages/ui/components/MediaControls/MediaButtons.tsx:63]:
```suggestion
```

File: packages/ui/components/MediaControls/MediaControls.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+    return function cleanup () {
+      document.removeEventListener('timeupdate', setTime)
+      document.removeEventListener('loadeddata', setData)
+      document.removeEventListener('ended', restartData)
+    }
+  }, [])
+
+  useEffect(() => {
+    console.log('fileref current')
```
Comment (gmarcos87) [inline packages/ui/components/MediaControls/MediaControls.tsx:54]: This log can be removed

File: packages/ui/components/VideoView/VideoView.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+export const VideoView: FunctionComponent<Props> = ({file}) => {
+  const [canPlay, handleCanPlay] = useState<boolean>(false)
+  const fileRef = useRef<HTMLVideoElement | null>(null)
+
+  useEffect(() => {
+    if (!fileRef) {
+      return
+    }
+
+    fileRef.current.addEventListener('canplay', () => {
```
Comment (gmarcos87) [inline packages/ui/components/VideoView/VideoView.tsx:19]: Clean the listeners on the return

File: packages/ui/components/VideoView/VideoView.tsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+      handleCanPlay(true)
+    })
+  }, [])
+
+  useEffect(() => {
+    handleCanPlay(false) 
+    fileRef.current.load()
+
+    fileRef.current.addEventListener('canplay', () => {
```
Comment (gmarcos87) [inline packages/ui/components/VideoView/VideoView.tsx:28]: Clean the listeners on the return

**Review by gmarcos87 — CHANGES_REQUESTED** [review]
I haven't run it yet, the changes are on a quick read of the code. I'll see how it works and leave you another comment if necessary.

**Comment by gmarcos87** [comment]
Hi @juandans01 If you can add the storybook views for the new components that would be great! If you can't now please create the ticket in the backlog so that it is documented as a pending task.

**Comment by juandans01** [comment]
Hi Marcos! I made the changes based on your feedback
let me know what you think :)

File: storybook/stories/FileView.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+
+storiesOf("FileView", module)
+  .add("ImageView", () => (
+    <div className="w-24 h-24">
+      <FileView file={FilesMokedData[0]}/>
+    </div>
+  ))
+  .add("AudioView", () => (
+    <div className="w-24 h24">
```
Comment (gmarcos87) [inline storybook/stories/FileView.stories.tsx:14]:
```suggestion
    <div className="w-24 h-24">
```
The player is not visible anyway. And the files in the mock are not audio or video, so you can't see the story correctly.

File: storybook/moked/files.ts [hunk 11]
```diff
@@ -31,22 +26,16 @@ export const FilesMokedData: ReportFile[] = [
...
-      name: "",
-    },
+    type: ReportFileType.AUDIO,
+    src: "",
     size: Math.random() * 10000000,
   },
   {
     id: "1",
     bucket: "",
+    src: "",
```
Comment (gmarcos87) [inline storybook/moked/files.ts:36]: We need to have an audio file to be able to check that it works.

**Review by gmarcos87 — CHANGES_REQUESTED** [review]
Done!, I just left some comments. Mainly that the storybook view is not working and I couldn't see how the player works. If you fix it and upload it, it would be approved immediately.

**Review by gmarcos87 — APPROVED** [review]

**Pull request merged**